src/target_list.rs

```diff
@@ -0,0 +1,23 @@
+use std::ptr;
+
+use ffi;
+use glib::translate::*;
+use {TargetEntry, TargetList};
+
+impl TargetList {
+    pub fn new(targets: &[TargetEntry]) -> Self {
+        let targets_glib: Vec<_> = targets
+                                       .iter()
+                                       .map(|e| e.to_glib_none().0)
+                                       .collect();
+        let t_ptr: *mut ffi::GtkTargetEntry = if targets_glib.len() > 0 {
```

Comment:
`.map(|e| e.to_glib_none().0)` will drop stash and collect bad pointers. See drag_dest_set